transport/http/outbound.go

```diff
@@ -121,35 +180,16 @@ func (o *outbound) createSpan(ctx context.Context, req *http.Request, treq *tran
 	ext.HTTPUrl.Set(span, req.URL.String())
 	ctx = opentracing.ContextWithSpan(ctx, span)
 
-	req.Header = applicationHeaders.ToHTTPHeaders(treq.Headers, nil)
-
 	tracer.Inject(
 		span.Context(),
 		opentracing.HTTPHeaders,
 		opentracing.HTTPHeadersCarrier(req.Header),
 	)
 
-	return ctx, span
+	return ctx, req, span
```

Comment:
this is a bit of a personal styling thing, pass by references mutations can obfuscate that mutations happen in a function.  I realize returning the req doesn't reaaaaaally do anything, it was more for callsite explicitness.  I can remove them if we want.